utils/RESTWalker.py

```diff
@@ -0,0 +1,194 @@
+                    self.templates[templatekey] = jinja2.Template(fh.read())
+
+    def _config_consumer(self, STUB, dirname, fnames):
+        for fname in fnames:
+            if fname.endswith('.json'):
+                with io.open(os.path.join(dirname, fname)) as fh:
+                    configkey = fname[:-len('.json')]
+                    self.raw_configs[configkey] = json.load(fh)
+
+    def _consume_directory(self, dirname, consumer):
+        os.path.walk(dirname, consumer, self)
+
+    def list_templates(self):
+        pp(self.templates.keys())
+
+    def list_raw_configs(self):
+        pp(self.raw_configs.keys())
+
+    def process_config(self, config_name):
+        raw_conf = self.raw_configs[config_name]
+        if 'kind' in raw_conf:
+            python_as_string = self._process_config_with_kind(raw_conf)
+        else:
+            raise Kindless(raw_conf)
+        return python_as_string
+
+    def _klassify(self, raw_klass):
+        caps = '_'.join([x.capitalize() for x in raw_klass.split('-')])
+        dotless = '_'.join([x.capitalize() for x in caps.split('.')])
+        return dotless
+
+    def _build_import_dicts(self, raw_conf, klass):
+        imports = []
+        selfLinkstart = raw_conf[u"selfLink"].partition("?")[0]
+        items = raw_conf[u"items"]
+        for item in items:
+            if item[u"reference"][u"link"].startswith(selfLinkstart):
+                tempuri = item[u"reference"][u"link"]
+                pre_qm = tempuri.partition("?")[0]
+                post_sl = pre_qm[len(selfLinkstart)+1:]
+                caps = self._klassify(post_sl)
+                imports.append({'OC': '.'+klass.lower(),
+                                'module': '.'+caps.lower(),
+                                'klass': caps})
+        return imports
+
+    def _format_org_collection(self, org_match, kind, raw_conf):
+        klass = org_match.group('OrgColl').capitalize()
+        OC_template = self.templates['OrganizingCollection']
+        import_dicts = self._build_import_dicts(raw_conf, klass)
+        config_dict = {'klass': klass,
+                       'kind': kind,
+                       'import_dicts': import_dicts}
+        OrgCollstr = OC_template.render(**config_dict)
+        imports_as_str = self.import_template.render(import_dicts=import_dicts)
+        imps_as_list = imports_as_str.split('\n')
+        imps_as_list.append(
+            'from f5.bigip.resource import OrganizingCollection')
+        imps_as_list.sort()
+        imports = '\n'.join(imps_as_list)
+        python_as_string = self.license_template.render()+imports+OrgCollstr
+        self.formatted_configs = {klass: {'Python_str': python_as_string,
+                                          'config_dict': config_dict,
+                                          'import_dicts': import_dicts}}
+        return python_as_string
+
+    def _format_resource(self, kind, raw_conf):
+        raw_string = kind.split(':')[-1][:-len('state')]
```

Comment:
This resource name is often the second element from the end? Is it always the case? And do you want to know if that's not the case?